langchain/src/agents/openai/output_parser.ts

```diff
@@ -0,0 +1,83 @@
+import type { OpenAI as OpenAIClient } from "openai";
+import {
+  AgentAction,
+  AgentFinish,
+  BaseMessage,
+  ChatGeneration,
+  Generation,
+  isBaseMessage,
+} from "../../schema/index.js";
+import { AgentActionOutputParser } from "../types.js";
+import { OutputParserException } from "../../schema/output_parser.js";
+
+/**
+ * Type that represents an agent action with an optional message log.
+ */
+export type FunctionsAgentAction = AgentAction & {
+  messageLog?: BaseMessage[];
+};
+
+export class OpenAIFunctionsAgentOutputParser extends AgentActionOutputParser {
+  lc_namespace = ["langchain", "agents", "openai"];
+
+  constructor() {
+    super(...arguments);
+  }
+
+  async parse(text: string): Promise<AgentAction | AgentFinish> {
+    throw new Error(
+      `OpenAIFunctionsAgentOutputParser can only parse messages.\nPassed input: ${text}`
+    );
+  }
+
+  async parseResult(generations: Generation[] | ChatGeneration[]) {
+    if ("message" in generations[0] && isBaseMessage(generations[0].message)) {
+      return this.parseAIMessage(generations[0].message);
+    }
+    throw new Error(
+      "parseResult on OpenAIFunctionsAgentOutputParser only works on ChatGeneration output"
```

Comment:
Why allow both as input here?

Reply:
Oversight on my part, I was just copypasta-ing the base class inputs.